k8s/apis/workflow/v1/activation_webhook.go

```diff
@@ -86,7 +86,7 @@ func (r *Activation) Default() {
 
 // TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
 
-//+kubebuilder:webhook:path=/validate-workflow-symphony-v1-activation,mutating=false,failurePolicy=fail,sideEffects=None,groups=workflow.symphony,resources=activations,verbs=create;update,versions=v1,name=mactivation.kb.io,admissionReviewVersions=v1
+//+kubebuilder:webhook:path=/validate-workflow-symphony-v1-activation,mutating=false,failurePolicy=fail,sideEffects=None,groups=workflow.symphony,resources=activations,verbs=create;update,versions=v1,name=vactivation.kb.io,admissionReviewVersions=v1
```

Comment:
Did you try the upgrade the scenario to see if the webhook name can be fixed seamlessly? @RemindD FYI.

Reply:
It's just because of using the same name in mutate webhook and validate webhook. I'm not fixing a typo now.

Reply:
Finished e2e test to verified the upgrade us seamlessly for the user.
```
{"level":"info","time":"2024-11-06T05:20:54.740Z","caller":"/k8s/apis/workflow/v1/activation_webhook.go:77","msg":"default","diagnostics":"{\"correlationId\":\"\",\"resourceId\":\"\",\"traceContext\":{\"spanId\":\"\",\"traceId\":\"\"}}","name":"hello-world-activation","namespace":"default","spec":{"campaign":"hello-world:v1","inputs":{"foo":"bar"}},"status":{}}
{"level":"info","time":"2024-11-06T05:20:54.740Z","caller":"/k8s/apis/workflow/v1/activation_webhook.go:82","msg":"default","diagnostics":"{\"correlationId\":\"\",\"resourceId\":\"\",\"traceContext\":{\"spanId\":\"\",\"traceId\":\"\"}}","name":"hello-world-activation","namespace":"default","spec.campaign":"hello-world:v1"}
{"level":"info","time":"2024-11-06T05:20:54.743Z","caller":"/k8s/apis/monitor/v1/utils.go:47","msg":"No diagnostics resource found for the namespace and edge location","namespace":"default","edgeLocation":""}
{"level":"info","time":"2024-11-06T05:20:54.743Z","caller":"/k8s/apis/workflow/v1/activation_webhook.go:126","msg":"validate update","diagnostics":"{\"correlationId\":\"\",\"resourceId\":\"\",\"traceContext\":{\"spanId\":\"\",\"traceId\":\"\"}}","name":"hello-world-activation","namespace":"default"}
{"level":"info","time":"2024-11-06T05:20:54.747Z","caller":"/k8s/controllers/workflow/activation_controller.go:53","msg":"Reconciling Activation","controller":"activation","controllerGroup":"workflow.symphony","controllerKind":"Activation","Activation":{"name":"hello-world-activation","namespace":"default"},"namespace":"default","name":"hello-world-activation","reconcileID":"31bd84e2-7ce4-47b2-87ee-c2918c4cd3b2","Name":"hello-world-activation","Namespace":"default"}
{"level":"info","time":"2024-11-06T05:20:54.747Z","caller":"/k8s/controllers/workflow/activation_controller.go:67","msg":"Activation status: Unknown State: 0","controller":"activation","controllerGroup":"workflow.symphony","controllerKind":"Activation","Activation":{"name":"hello-world-activation","namespace":"default"},"namespace":"default","name":"hello-world-activation","reconcileID":"31bd84e2-7ce4-47b2-87ee-c2918c4cd3b2"}
{"level":"info","time":"2024-11-06T05:20:54.752Z","caller":"/k8s/controllers/workflow/activation_controller.go:53","msg":"Reconciling Activation","controller":"activation","controllerGroup":"workflow.symphony","controllerKind":"Activation","Activation":{"name":"hello-world-activation","namespace":"default"},"namespace":"default","name":"hello-world-activation","reconcileID":"403c252d-0805-4399-8b0d-f961305acc6f","Name":"hello-world-activation","Namespace":"default"}
{"level":"info","time":"2024-11-06T05:20:54.752Z","caller":"/k8s/controllers/workflow/activation_controller.go:67","msg":"Activation status: Running","controller":"activation","controllerGroup":"workflow.symphony","controllerKind":"Activation","Activation":{"name":"hello-world-activation","namespace":"default"},"namespace":"default","name":"hello-world-activation","reconcileID":"403c252d-0805-4399-8b0d-f961305acc6f"}
{"level":"info","time":"2024-11-06T05:20:54.761Z","caller":"/k8s/apis/workflow/v1/activation_webhook.go:77","msg":"default","diagnostics":"{\"correlationId\":\"\",\"resourceId\":\"\",\"traceContext\":{\"spanId\":\"\",\"traceId\":\"\"}}","name":"hello-world-activation","namespace":"default","spec":{"campaign":"hello-world:v1","inputs":{"foo":"bar"}},"status":{"status":9994,"statusMessage":"Running","updateTime":"2024-11-06T05:20:54Z","stageHistory":[{"stage":"mock","inputs":null,"outputs":{"__namespace":"default"},"status":9994,"statusMessage":"Running","isActive":true}]}}
{"level":"info","time":"2024-11-06T05:20:54.761Z","caller":"/k8s/apis/workflow/v1/activation_webhook.go:82","msg":"default","diagnostics":"{\"correlationId\":\"\",\"resourceId\":\"\",\"traceContext\":{\"spanId\":\"\",\"traceId\":\"\"}}","name":"hello-world-activation","namespace":"default","spec.campaign":"hello-world:v1"}
{"level":"info","time":"2024-11-06T05:20:54.763Z","caller":"/k8s/apis/monitor/v1/utils.go:47","msg":"No diagnostics resource found for the namespace and edge location","namespace":"default","edgeLocation":""}
{"level":"info","time":"2024-11-06T05:20:54.763Z","caller":"/k8s/apis/workflow/v1/activation_webhook.go:126","msg":"validate update","diagnostics":"{\"correlationId\":\"\",\"resourceId\":\"\",\"traceContext\":{\"spanId\":\"\",\"traceId\":\"\"}}","name":"hello-world-activation","namespace":"default"}
{"level":"info","time":"2024-11-06T05:20:54.766Z","caller":"/k8s/controllers/workflow/activation_controller.go:53","msg":"Reconciling Activation","controller":"activation","controllerGroup":"workflow.symphony","controllerKind":"Activation","Activation":{"name":"hello-world-activation","namespace":"default"},"namespace":"default","name":"hello-world-activation","reconcileID":"439074bf-5820-4a9f-8e8a-07d7effa64a8","Name":"hello-world-activation","Namespace":"default"}
{"level":"info","time":"2024-11-06T05:20:54.766Z","caller":"/k8s/controllers/workflow/activation_controller.go:67","msg":"Activation status: Running","controller":"activation","controllerGroup":"workflow.symphony","controllerKind":"Activation","Activation":{"name":"hello-world-activation","namespace":"default"},"namespace":"default","name":"hello-world-activation","reconcileID":"439074bf-5820-4a9f-8e8a-07d7effa64a8"}
{"level":"info","time":"2024-11-06T05:20:55.131Z","caller":"/k8s/controllers/workflow/activation_controller.go:53","msg":"Reconciling Activation","controller":"activation","controllerGroup":"workflow.symphony","controllerKind":"Activation","Activation":{"name":"hello-world-activation","namespace":"default"},"namespace":"default","name":"hello-world-activation","reconcileID":"1cdd59ff-a5fb-409a-b529-0acf5a599e42","Name":"hello-world-activation","Namespace":"default"}
{"level":"info","time":"2024-11-06T05:20:55.131Z","caller":"/k8s/controllers/workflow/activation_controller.go:67","msg":"Activation status: Done","controller":"activation","controllerGroup":"workflow.symphony","controllerKind":"Activation","Activation":{"name":"hello-world-activation","namespace":"default"},"namespace":"default","name":"hello-world-activation","reconcileID":"1cdd59ff-a5fb-409a-b529-0acf5a599e42"}
{"level":"info","time":"2024-11-06T05:21:17.510Z","caller":"/k8s/apis/monitor/v1/utils.go:47","msg":"No diagnostics resource found for the namespace and edge location","namespace":"default","edgeLocation":""}
{"level":"info","time":"2024-11-06T05:21:17.510Z","caller":"/k8s/apis/workflow/v1/campaign_webhook.go:190","msg":"validate delete","diagnostics":"{\"correlationId\":\"\",\"resourceId\":\"\",\"traceContext\":{\"spanId\":\"\",\"traceId\":\"\"}}","name":"hello-world-v-v1","namespace":"default"}
{"level":"info","time":"2024-11-06T05:21:27.287Z","caller":"/k8s/controllers/workflow/activation_controller.go:53","msg":"Reconciling Activation","controller":"activation","controllerGroup":"workflow.symphony","controllerKind":"Activation","Activation":{"name":"hello-world-activation","namespace":"default"},"namespace":"default","name":"hello-world-activation","reconcileID":"7ed6ce85-9e01-4c1f-b5f9-3859dae1de27","Name":"hello-world-activation","Namespace":"default"}
{"level":"error","time":"2024-11-06T05:21:27.287Z","caller":"/k8s/controllers/workflow/activation_controller.go:58","msg":"unable to fetch Activation","controller":"activation","controllerGroup":"workflow.symphony","controllerKind":"Activation","Activation":{"name":"hello-world-activation","namespace":"default"},"namespace":"default","name":"hello-world-activation","reconcileID":"7ed6ce85-9e01-4c1f-b5f9-3859dae1de27","error":"Activation.workflow.symphony \"hello-world-activation\" not found","stacktrace":"gopls-workspace/controllers/workflow.(*ActivationReconciler).Reconcile\n\t/k8s/controllers/workflow/activation_controller.go:58\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.5/pkg/internal/controller/controller.go:114\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.5/pkg/internal/controller/controller.go:311\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.5/pkg/internal/controller/controller.go:261\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.5/pkg/internal/controller/controller.go:222"}

```